controllers/storagecluster/provider_server.go

```diff
@@ -434,3 +444,62 @@ func RandomString(l int) string {
 
 	return string(bytes)
 }
+
+func getOnboardingJobObject(instance *ocsv1.StorageCluster) *batchv1.Job {
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingJobName,
+			Namespace: instance.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: onboardingJobName,
+					Containers: []corev1.Container{
+						{
+							Name:    onboardingJobName,
+							Image:   os.Getenv(onboardingSecretGeneratorImage),
+							Command: []string{"/usr/local/bin/onboarding-secret-generator"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "OPERATOR_NAMESPACE",
+									Value: instance.Namespace,
+								},
```

Comment:
- if this env wasn't present main controller wouldn't start and so you can directly getenv
```suggestion
								{
									Name:  util.OperatorNamespaceEnvVar,
									Value: os.Getenv(util.OperatorNamespaceEnvVar),
								},
								
								
```